packages/gateway/src/manager.ts

```diff
@@ -170,24 +191,21 @@ export function createGatewayManager(options: CreateGatewayManagerOptions): Gate
           })
         })
       },
-      async updateGuildsShardId(guildIds, shardId) {
-        gateway.logger.debug(
-          `[Resharding] Updating the following guild ids shard to #${shardId}: ${guildIds.join(', ')}. Override this function to update your cache if you need to.`,
-        )
-      },
+      async updateGuildsShardId(_guildIds, _shardId) {},
       async shardIsPending(shard) {
         // Save this in pending at the moment, until all shards are online
         gateway.resharding.pendingShards.set(shard.id, shard)
         gateway.logger.warn(`[Resharding] Shard #${shard.id} is now pending`)
+
         // Check if all shards are now online.
-        if (gateway.lastShardId - gateway.firstShardId + 1 > gateway.resharding.pendingShards.size) return
+        if (gateway.lastShardId - gateway.firstShardId >= gateway.resharding.pendingShards.size) return
+
         gateway.logger.warn(`[Resharding] All shards are now online.`)
 
         // New shards start processing events
-        for (const shard of gateway.resharding.pendingShards.values()) {
+        for (const shard of gateway.resharding.shards.values()) {
```

Comment:
@Fleny113 do we need to undo this change too? You said its not instantly populated

Reply:
no, it isn't instantly populated, but by the time this line runs they are the same

What i meant in commit https://github.com/discordeno/discordeno/pull/3792/commits/59dcf503b6b93eacaf57d4a2be36a8741dfb6628 is that in the if above that prevents stuff like this for-of to run when that runs `shards` is not fully populated, but if the if check has passed then they are the same and this doesn't need to be reverted, it is just easier (imo) to understand with the iteration on the `shards` collection rather than the `pendingShards` collection